src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -98,6 +101,13 @@
  */
 public final class StringConcatFactory {
 
+    private static final int HIGH_ARITY_THRESHOLD;
+
+    static {
+        String highArity = VM.getSavedProperty("java.lang.invoke.StringConcat.highArityThreshold");
+        HIGH_ARITY_THRESHOLD = Integer.parseInt(highArity != null ? highArity : "20");
```

Comment:
Maybe write this as `= (highArity != null) ? Integer.parseInt(highArity) : 20` to help interpreter a bit.